packages/app/e2e/config.e2e.js

```diff
@@ -39,6 +39,13 @@ describe('config', () => {
       await NativeModules.RNFBAppModule.preferencesClearAll();
     });
 
+    // NOTE: "preferencesClearAll" clears Firestore settings. Set DB as emulator again.
```

Comment:
That is fiendish + subtle by the way, nice catch

Reply:
@russellwheatley this caused me a problem that I just figured out, at least part way

- I have never been able to run the firestore e2e tests as is locally after this change
- I presume then that there is a difference between firestore in the cloud and firestore emulator that impacts the tests
- I check the errors I get and they all have to do with errors not being thrown into the sinon.spy when we expect them to be
- I check the firestore-debug.log from the emulator and I see:

```
[warn] ⚠  firestore: The emulator will default to allowing all reads and writes. Learn more about this option: https://firebase.google.com/docs/emulator-suite/install_and_configure#security_rules_configuration. {"metadata":{"emulator":{"name":"firestore"},"message":"The emulator will default to allowing all reads and writes. Learn more about this option: https://firebase.google.com/docs/emulator-suite/install_and_configure#security_rules_configuration."}}

```

I infer that the firestore emulator is not running with the same security rules somehow? I don't have insight into the firebase-in-the-cloud rules (they are not in the firestore.json that exists in the tests project) so I don't know what the difference could be.

Question: do you have access to the firestore security rules for the cloud project? Could you paste them in here?

But I have to revert this change at least temporarily in order to pass tests

Reply:
@mikehardy Woops, just spotted this. The firestore rules are [here](https://github.com/invertase/react-native-firebase/blob/master/.github/workflows/scripts/firestore.rules). When we run this [script](https://github.com/invertase/react-native-firebase/blob/master/.github/workflows/scripts/start-firestore-emulator.sh) in the [CI](https://github.com/invertase/react-native-firebase/blob/master/.github/workflows/tests_e2e.yml#L54), it ought to use those rules 🤔

Reply:
Perfect thanks for the pointers! I'll see what I can do

Reply:
@russellwheatley just to close the loop - Mike D pointed out that it might be exactly how the script was started, specifically it requires you start it in the correct working directory, if you start it outside the directory with the rules they do not apply and you see the problem I saw. I fixed up the CI / package.json scripts so it will be reliable for everyone, and you may already have been aware but just in case you weren't: careful with the working directory of the firebase emulator :-)